src/cow-react/modules/utm/hooks.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react'
+import { useLocation } from 'react-router-dom'
+import { UtmParams } from './types'
+import { useSetAtom } from 'jotai'
+import { utmAtom } from './state'
+
+function getUtcParams(searchParams: URLSearchParams): UtmParams {
+  const utcSource = searchParams.get('utm_source') || undefined
+  const utcMedium = searchParams.get('utm_medium') || undefined
+  const utcCampaign = searchParams.get('utm_campaign') || undefined
+  const utcContent = searchParams.get('utm_content') || undefined
+  const utcTerm = searchParams.get('utm_term') || undefined
+
```

Comment:
Shouldn't we also delete the params after loading them?
That's what's done for the other url params